core/templates/components/graph-services/graph-layout.service.spec.ts

```diff
@@ -0,0 +1,1346 @@
+    }
+
+    // Check if the returned augmented links have a bezier curve
+    // which is equal to the expected value.
+    expect(returnedBezierCurveValues).toEqual(expectedBezierCurveValues);
+  });
+
+  it('should return undefined when source and target nodes overlap' +
+    ' while processing augmented links', () => {
+    // The nodes State1 and State2 overlap as State1.xLabel === State2.xLabel
+    // and State1.yLabel === State2.yLabel .
+    let nodeData = {
+      State1: {
+        depth: 0,
+        offset: 0,
+        reachable: true,
+        x0: 0.07250000000000001,
+        y0: 0.15333333333333335,
+        xLabel: 0.1625,
+        yLabel: 0.23333333333333334,
+        id: 'State1',
+        label: 'State1',
+        height: 0.16,
+        width: 0.18000000000000002,
+        reachableFromEnd: false
+      },
+      State2: {
+        depth: 1,
+        offset: 0,
+        reachable: true,
+        x0: 0.07250000000000001,
+        y0: 0.42000000000000004,
+        xLabel: 0.1625,
+        yLabel: 0.23333333333333334,
+        id: 'State2',
+        label: 'State2',
+        height: 0.16,
+        width: 0.18000000000000002,
+        reachableFromEnd: false
+      }
+    };
+
+    let links = [
+      {
+        source: 'State1',
+        target: 'State2'
+      }
+    ];
+
+    expect(sgls.getAugmentedLinks(nodeData, links)).toBeUndefined();
+  });
+
+  it('should get correct graph width and height', () => {
+    let nodeData = {
+      State1: {
+        depth: 1,
+        offset: 0,
+        reachable: true,
+        x0: 0.07250000000000001,
+        y0: 0.42000000000000004,
+        xLabel: 0.1625,
+        yLabel: 0.5,
+        id: 'State1',
+        label: 'State1',
+        height: 0.16,
+        width: 0.18000000000000002,
+        reachableFromEnd: false
+      },
+      Introduction: {
+        depth: 0,
+        offset: 0,
+        reachable: true,
+        x0: 0.07250000000000001,
+        y0: 0.15333333333333335,
+        xLabel: 0.1625,
+        yLabel: 0.23333333333333334,
+        id: 'Introduction',
+        label: 'Introduction',
+        height: 0.16,
+        width: 0.18000000000000002,
+        reachableFromEnd: false
+      },
+      End: {
+        depth: 2,
+        offset: 0,
+        reachable: true,
+        x0: 0.07250000000000001,
+        y0: 0.6866666666666666,
+        xLabel: 0.1625,
+        yLabel: 0.7666666666666666,
+        id: 'End',
+        label: 'End',
+        height: 0.16,
+        width: 0.18000000000000002,
+        reachableFromEnd: false
+      },
+      State2: {
+        depth: 1,
+        offset: 1,
+        reachable: true,
+        x0: 0.29750000000000004,
+        y0: 0.42000000000000004,
+        xLabel: 0.3875,
+        yLabel: 0.5,
+        id: 'State2',
+        label: 'State2',
+        height: 0.16,
+        width: 0.18000000000000002,
+        reachableFromEnd: false
+      }
+    };
+
+    let graphWidthUpperBoundInPixel = sgls.getGraphWidth(
```

Comment:
Please, read the previous comment carefully -- your replacement doesn't follow any of the structures given there. "In Pixel" is not correct because normally measurements are pluralized. Either of the variants I suggested are OK.

If a reviewer suggests something, and you do something different with no explanation or acknowledgement, this is what will make your PR reviews take longer, because of the additional back-and-forth.

Reply:
Changed the variable names